File: stack & queues/Stack.java

public class Stack {

    Node head;

    Stack() {
        this.head = null;
    }

    class Node {

        Node next;
        int data;

        Node(int data) {
            this.next = null;
            this.data = data;
        }
    }

    public Boolean isEmpty() {
        return head == null;
    }

    public void push(int data) {
        Node newNode = new Node(data);
        if (isEmpty()) {
            head = newNode;
            return;
        }
        newNode.next = head;
        head = newNode;
    }

    public int pop() {
        if (isEmpty()) {
            return -1;
        }
        int data = head.data;
        head = head.next;
        return data;

    }

    public int peek() {
        if (isEmpty()) {
            return -1;
        }
        return head.data;
    }

    public static void main(String args[]) {
        Stack s = new Stack();
        s.push(10);
        s.push(20);
        s.push(30);
        while (!s.isEmpty()) {
            System.out.println(s.pop() + " ");
        }

    }

}

File: linear data structures/stack/stack.py
class Stack:
    def __init__(self):
        self.stack=[]
        self.top=-1

    def push(self,data):
        self.top+=1
        self.stack.append(data)

    def pop(self):
        data=self.stack[self.top]
        self.top-=1
        self.stack.pop()
        return data

    def peek(self):
        return self.stack[self.top]

if __name__=="__main__":
    stack=Stack()
    stack.push(1)
    stack.push(100)
    print(stack.stack)
    stack.push(2)
    stack.push(3)
    stack.push(3)
    stack.push(99)
    print(stack.stack)
    stack.pop()
    stack.pop()
    print(stack.stack)


